streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -921,6 +929,9 @@ private boolean close(final long timeoutMs) {
                 setState(State.NOT_RUNNING);
             }, "kafka-streams-close-thread");
 
+            if (RecordingLevel.forName(config.getString(METRICS_RECORDING_LEVEL_CONFIG)) == RecordingLevel.DEBUG) {
```

Comment:
IIUC, `rocksDBMetricsRecordingTriggerThread` is either going to be `null` because it didn't meet the criteria for creating the thread or it is going to be non null because we did create it. Given that, checking for `null` is a nicer way to determine if we need to shutdown because as long as we have a thread, regardless of the condition that created it (e.g. `RecordingLevel.DEBUG`), we should shut it down. This is also safer if the creation conditions ever change and we forget to update them here.

Reply:
Agreed!